wasi.go

```diff
@@ -69,33 +68,31 @@ func (c *WASIConfig) WithPreopens(preopens map[string]wasi.FS) *WASIConfig {
 
 // WASISnapshotPreview1 are functions importable as the module name wasi.ModuleSnapshotPreview1
 func WASISnapshotPreview1() *Module {
-	return WASISnapshotPreview1WithConfig(&WASIConfig{})
-}
-
-// WASISnapshotPreview1WithConfig are functions importable as the module name wasi.ModuleSnapshotPreview1
-func WASISnapshotPreview1WithConfig(c *WASIConfig) *Module {
-	cfg := newConfig(c) // safe copy of config
-	_, fns := internalwasi.SnapshotPreview1Functions(cfg)
+	_, fns := internalwasi.SnapshotPreview1Functions()
 	m, err := internalwasm.NewHostModule(wasi.ModuleSnapshotPreview1, fns)
 	if err != nil {
 		panic(fmt.Errorf("BUG: %w", err))
 	}
 	return &Module{name: wasi.ModuleSnapshotPreview1, module: m}
 }
 
-func newConfig(c *WASIConfig) *internalwasi.Config {
-	cfg := internalwasi.NewConfig()
+func newSystemContext(c *WASIConfig) (*internalwasm.SystemContext, error) {
```

Comment:
Specifically #394 will rename the WASIConfig type to ModuleConfig or SystemConfig and pass that to instantiate. This will prevent having to swap references later, and also ensure error handling happens before any instantiation might occur.

Reply:
I'm waiting to do that as it is a bigger break than this change, and we might as well fix wasi.FS -> fs.FS while in the same change.